cmd/lakefs/cmd/run.go

```diff
@@ -231,12 +254,17 @@ var runCmd = &cobra.Command{
 			auth.NewBuiltinAuthenticator(authService),
 		}
 
-		if cfg.Auth.LDAP != nil {
-			logger.
-				WithField("feature", "LDAP").
-				Warn("Enabling LDAP on lakeFS server, but this functionality is deprecated")
-			middlewareAuthenticator = append(middlewareAuthenticator, newLDAPAuthenticator(cfg.Auth.LDAP, authService))
+		externalAuthenticator, err := newExternalBasicAuthenticator(cfg, authService, logger)
+
+		if err != nil {
+			fmt.Printf("%s: %s\n", "failed to create external authenticator", err)
```

Comment:
Please exit lakeFS from the main program.  This function could just return an error, which would let us wrap it with additional information about why the LDAP authenticator got called.

Also if we did that then we would not be writing error messages to standard output as we end up doing here (although that is of course easy to fix at this place).

Reply:
I'm not quite sure what you mean, can you take a look at the change I added? If not that then please elaborate. 
Also, assuming I understand you correctly notice I applied the same change to `allowForeign, err := cmd.Flags().GetBool(mismatchedReposFlagName)` in the same file as well.